pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -234,6 +236,7 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 
 	if !r.upgradeMode && !init && knownHcoVersion != r.ownVersion {
 		r.upgradeMode = true
+		req.componentUpgradeDone = true
```

Comment:
if we detect the upgrade mode we track it with:
r.upgradeMode = true
but then
req.componentUpgradeDone = true

componentUpgradeDone should't we have the opposite on componentUpgradeDone because we know that somoething can also happen to other components?